File: 30.TaggedPointer/30.TaggedPointer/ViewController.m
//
//  ViewController.m
//  30.TaggedPointer
//
//  Created by jackfrow on 2020/2/6.
//  Copyright © 2020 jackfrow. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
   
       NSNumber *number1 = @1;
       NSNumber *number2 = @2;
       NSNumber *number3 = @3;
       NSNumber *numberFFFF = @(0xFFFF);
       
       NSLog(@"number1 pointer is %p", number1);
       NSLog(@"number2 pointer is %p", number2);
       NSLog(@"number3 pointer is %p", number3);
       NSLog(@"numberffff pointer is %p", numberFFFF);
    
}


@end

File: 24.CalayerTutorial/24.CalayerTutorial/ViewController.swift
//
//  ViewController.swift
//  24.CalayerTutorial
//
//  Created by yy on 2019/9/17.
//  Copyright © 2019 Jackfrow. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var viewForLayer: UIView!
    
    var layer: CALayer {
        return viewForLayer.layer
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
   
        setUplayer()
        
    }
    
    func setUplayer()  {
        layer.backgroundColor = UIColor.blue.cgColor
        layer.borderWidth = 100.0
        layer.borderColor = UIColor.red.cgColor
        layer.shadowOpacity = 0.7
        layer.shadowRadius = 10.0
        layer.contents = UIImage(named: "star")?.cgImage
        layer.contentsGravity = CALayerContentsGravity.center
    }


}

